src/OpenIddict.EntityFramework/Stores/OpenIddictApplicationStore.cs

```diff
@@ -0,0 +1,140 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace OpenIddict {
+    /// <summary>
+    /// Provides methods allowing to manage the applications stored in a database.
+    /// </summary>
+    /// <typeparam name="TApplication">The type of the Application entity.</typeparam>
+    /// <typeparam name="TContext">The type of the Entity Framework database context.</typeparam>
+    /// <typeparam name="TKey">The type of the entity primary keys.</typeparam>
+    public class OpenIddictApplicationStore<TApplication, TContext, TKey> : IOpenIddictApplicationStore<TApplication>
+        where TApplication : OpenIddictApplication<TKey>
+        where TContext : DbContext
+        where TKey : IEquatable<TKey> {
+        public OpenIddictApplicationStore(TContext context) {
+            Context = context;
+        }
+
+        /// <summary>
+        /// Gets the database context associated with the current store.
+        /// </summary>
+        public virtual TContext Context { get; }
```

Comment:
Are there any specific reasons for make this public? otherwise can we make this protected?